python/lsst/sims/coordUtils/LsstCameraUtils.py

```diff
@@ -0,0 +1,427 @@
+            name_list.append(chip_name)
+
+    x_pix_list = np.array(x_pix_list)
+    y_pix_list = np.array(y_pix_list)
+
+    x_pup_list, y_pup_list = pupilCoordsFromPixelCoords(x_pix_list,
+                                                        y_pix_list,
+                                                        name_list,
+                                                        camera=_lsst_camera)
+    center_x = np.zeros(n_chips, dtype=float)
+    center_y = np.zeros(n_chips, dtype=float)
+    extent = np.zeros(n_chips, dtype=float)
+    final_name = []
+    for ix_ct in range(n_chips):
+        ix = ix_ct*4
+        chip_name = name_list[ix]
+        xx = 0.25*(x_pup_list[ix] + x_pup_list[ix+1] +
+                   x_pup_list[ix+2] + x_pup_list[ix+3])
+
+        yy = 0.25*(y_pup_list[ix] + y_pup_list[ix+1] +
+                   y_pup_list[ix+2] + y_pup_list[ix+3])
+
+        dx = 0.25*np.array([np.sqrt(np.power(xx-x_pup_list[ix+ii], 2) +
+                                    np.power(yy-y_pup_list[ix+ii], 2)) for ii in range(4)]).sum()
+
+        center_x[ix_ct] = xx
+        center_y[ix_ct] = yy
+        extent[ix_ct] = dx
+        final_name.append(chip_name)
+
+    final_name = np.array(final_name)
+
+    _lsst_pupil_coord_map = {}
+    _lsst_pupil_coord_map['name'] = final_name
+    _lsst_pupil_coord_map['xx'] = center_x
+    _lsst_pupil_coord_map['yy'] = center_y
+    _lsst_pupil_coord_map['dp'] = extent
+
+
+def _findDetectorsListLSST(cameraPointList, detectorList, allow_multiple_chips=False):
+    """!Find the detectors that cover a list of points specified by x and y coordinates in any system
+
+    This is based one afw.camerGeom.camera.findDetectorsList.  It has been optimized for the LSST
+    camera in the following way:
+
+        - it accepts a limited list of detectors to check in advance (this list should be
+          constructed by comparing the pupil coordinates in question and comparing to the
+          pupil coordinates of the center of each detector)
+
+       - it will stop looping through detectors one it has found one that is correct (the LSST
+         camera does not allow an object to fall on more than one detector)
+
+    @param[in] cameraPointList  a list of cameraPoints in PUPIL coordinates
+
+    @param[in] detecorList is a list of lists.  Each row contains the detectors that should be searched
+    for the correspdonding cameraPoint
+
+    @param [in] allow_multiple_chips is a boolean (default False) indicating whether or not
+    this method will allow objects to be visible on more than one chip.  If it is 'False'
+    and an object appears on more than one chip, only the first chip will appear in the list of
+    chipNames but NO WARNING WILL BE EMITTED.  If it is 'True' and an object falls on more than one
+    chip, a list of chipNames will appear for that object.
+
+    @return outputNameList is a numpy array of the names of the detectors
+    """
+
+    global _lsst_camera
```

Comment:
I like avoiding globals by making the object an attribute of the function, but since `_lsst_camera` gets used other places too a global makes more sense.